### PR TITLE
Change `StandardScaler` to the `SquashScaler` for the `tabular_pipline`

### PR DESCRIPTION
**Pull request by dierickxsimon**
Changing the scaler used for linear models from StandardScaler to SquashingScaler.

### CODE REVIEW

**Comment by rcap107** [comment]
Thanks a lot @dierickxsimon! 

In the original issue there was a discussion on the value of `max_absolute_value`, so I'm paging @GaelVaroquaux or @dholzmueller to ask what should be used here

**Comment by dholzmueller** [comment]
I would say 5 or 10 might be good for `max_absolute_value`? maybe go conservative and use 10?

**Comment by GaelVaroquaux** [comment]
I'd rather go conversative

File: CHANGES.rst [hunk 1]
```diff
@@ -14,6 +14,9 @@ New features
 
 Changes
 -------
+- The :func: `tabular_pipeline` uses a :class:`SquashingScaler` instead of a
+  :class:`StandardScaler` for centering and standard scaling numerical features in the.
```
Comment (rcap107) [inline CHANGES.rst]:
```suggestion
  :class:`StandardScaler` for centering and scaling numerical features 
  when linear models are used. 
```

File: skrub/_tabular_pipeline.py [hunk 2]
```diff
@@ -99,6 +100,11 @@ def tabular_pipeline(estimator, *, n_jobs=None):
         The high cardinality encoder has been changed from
         :class:`~skrub.MinHashEncoder` to :class:`~skrub.StringEncoder`.
 
+    .. versionchanged:: 0.7.0
+        The :class:`~skrub.SquashingScaler` with `max_absolute_value=10` is now used instead of
+        :class:`~sklearn.preprocessing.StandardScaler` for centering and scaling
+        numerical features.
+
```
Comment (rcap107) [inline skrub/_tabular_pipeline.py:107]:
```suggestion
        :class:`~sklearn.preprocessing.StandardScaler` for centering and scaling
        numerical features when using linear models. 

```

File: skrub/_tabular_pipeline.py [hunk 3]
```diff
@@ -247,7 +253,7 @@ def tabular_pipeline(estimator, *, n_jobs=None):
     - A :obj:`~sklearn.impute.SimpleImputer`, as the
       :obj:`~sklearn.linear_model.LogisticRegression` cannot handle missing values.
 
-    - A :obj:`~sklearn.preprocessing.StandardScaler` for centering and standard scaling
+    - A :obj:`~skrub.SquashingScaler` for centering and standard scaling
```
Comment (rcap107) [inline skrub/_tabular_pipeline.py]:
```suggestion
    - A :obj:`~skrub.SquashingScaler` for centering and scaling
```

**Comment by dholzmueller** [comment]
I could relatively easily run some benchmarks to check different max_absolute_value with LinearModel. I'd prefer to do it in ~ two weeks though.

**Comment by rcap107** [comment]
> I could relatively easily run some benchmarks to check different max_absolute_value with LinearModel. I'd prefer to do it in ~ two weeks though.

I think we can wait until then without problem

**Comment by rcap107** [comment]
@dholzmueller do you think you could run the benchmark this week? otherwise we can always set the value to 10 and open a new pr later to set a different value once we have the results

**Comment by dholzmueller** [comment]
Yes, I plan to do it this week, sometime after my deadline (which is tomorrow).

**Comment by rcap107** [comment]
> Yes, I plan to do it this week, sometime after my deadline (which is tomorrow).

That's great, thanks a lot

**Comment by dholzmueller** [comment]
I ran linear regression with tuning and cross-validation ensembling on TALENT-regression (only applying the scalers to numerical features). 
- The SquashingScaler is better than standardization, achieving win-rates of >70% vs standardization but only small improvements <2% in terms of average scores.
- small clipping values (3, 5) are (significantly) better in terms of win-rates / ranks / normalized scores, but clipping at 10 is better on average scores because it's much better on few datasets.

I could also run classification experiments with logistic regression easily (but it might take a few hours).

**Comment by dholzmueller** [comment]
For binary classification on TALENT with cross-entropy loss, I also tested 3, 5, and 10. Smaller values are better in all aggregations (win-rate, normalized scores, ranks, average error, etc.). For all values, the SquashingScaler is better than standardization. 

Based on this, I would say we should probably take a smaller value like 3 or 5 unless we want to be good on the few datasets where having a linear function is important.

**Comment by rcap107** [comment]
I think we can go with 5, the `tabular_pipeline` is not intended to be the best in all situations, and I think that a value that is better in most scenarios is what we should go for here.

As I've mentioned IRL, @dholzmueller it would be great if we could put out a short post on how we chose the default value to add to the skrub-materials website. 

@dierickxsimon I think we can merge after setting the value to 5

**Comment by dholzmueller** [comment]
Here is a draft for the post, can you fill in title / docs references or whatever is needed?

To compare the SquashingScaler to the StandardScaler for linear/logistic regression, we ran them (with `max_abs_value` in {3, 5, 10} for the SquashingScaler) on a version of the [TALENT benchmark](https://arxiv.org/abs/2407.00956) using [pytabkit](https://github.com/dholzmueller/pytabkit). We noticed the following trends:
- Across aggregation strategies like average score, average normalized score, average rank, win-rate, the `SquashingScaler` performed better than the `StandardScaler` for all values of `max_abs_value`. Typically, variants of the `SquashingScaler` won on around 70%-75% of datasets vs the `StandardScaler`, though often only by a small margin on the order of 1% error reduction.
- Smaller values of `max_abs_value` were often better, with 3 being the best on classification. For regression, smaller values were better in terms of win-rates, average normalized scores, and average ranks, but there were a few datasets where `max_abs_value=10` achieved a much lower error. Perhaps this is because the ground-truth function on the dataset was close to linear.
Based on the observations above, we chose `max_abs_value=5` as a compromise.

Details: We used RMSE for regression and cross-entropy for classification, inner 8-fold cross-validation with ensembling, tuning on the [TabRepo](https://arxiv.org/abs/2311.02971)/[TabArena](https://arxiv.org/abs/2506.16791) search space (includes L1 or L2 regularization), 60%-40% train-test split, 2 repeats per dataset.

**Comment by rcap107** [comment]
> Here is a draft for the post, can you fill in title / docs references or whatever is needed?
> 
> To compare the SquashingScaler to the StandardScaler for linear/logistic regression, we ran them (with `max_abs_value` in {3, 5, 10} for the SquashingScaler) on a version of the [TALENT benchmark](https://arxiv.org/abs/2407.00956) using [pytabkit](https://github.com/dholzmueller/pytabkit?rgh-link-date=2025-10-16T08%3A43%3A19Z). We noticed the following trends:
> 
> * Across aggregation strategies like average score, average normalized score, average rank, win-rate, the `SquashingScaler` performed better than the `StandardScaler` for all values of `max_abs_value`. Typically, variants of the `SquashingScaler` won on around 70%-75% of datasets vs the `StandardScaler`, though often only by a small margin on the order of 1% error reduction.
> * Smaller values of `max_abs_value` were often better, with 3 being the best on classification. For regression, smaller values were better in terms of win-rates, average normalized scores, and average ranks, but there were a few datasets where `max_abs_value=10` achieved a much lower error. Perhaps this is because the ground-truth function on the dataset was close to linear.
>   Based on the observations above, we chose `max_abs_value=5` as a compromise.
> 
> Details: We used RMSE for regression and cross-entropy for classification, inner 8-fold cross-validation with ensembling, tuning on the [TabRepo](https://arxiv.org/abs/2311.02971)/[TabArena](https://arxiv.org/abs/2506.16791) search space (includes L1 or L2 regularization), 60%-40% train-test split, 2 repeats per dataset.

Thanks a lot, this will end on the skrub-materials website, so I'll move the draft there and ping you when it's ready
